src/System.Runtime.Extensions/src/System/IO/Path.cs

```diff
@@ -356,41 +360,42 @@ private static string CombineNoChecks(string path1, string path2, string path3)
                 'q', 'r', 's', 't', 'u', 'v', 'w', 'x',
                 'y', 'z', '0', '1', '2', '3', '4', '5'};
 
-        private static string ToBase32StringSuitableForDirName(byte[] buff)
+        private static unsafe void GetBase32CharsSuitableForDirName(byte* bytes, int byteCount, char* chars, int charCount)
         {
-            // This routine is optimised to be used with buffs of length 20
-            Debug.Assert(((buff.Length % 5) == 0), "Unexpected hash length");
+            Debug.Assert(bytes != null);
+            Debug.Assert(chars != null);
+            // This routine is optimized to be used with bytes of length 10 and chars of length 16.
+            Debug.Assert((byteCount >= 5) && ((byteCount % 5) == 0), $"Unexpected {nameof(byteCount)}");
+            Debug.Assert((charCount >= 8) && ((charCount % 8) == 0), $"Unexpected {nameof(charCount)}");
```

Comment:
Since this is only called from the one location, should we just assert what's called out in the comment rather than the more complicated asserts currently there?